torch/onnx/symbolic_opset11.py

```diff
@@ -299,18 +294,24 @@ def index_put(
         values = opset9.expand(g, values, values_shape, None)
     values = symbolic_helper._reshape_helper(g, values, values_shape)
 
-    dtype = self.type().scalarType()
-    if dtype is not None and dtype != values.type().scalarType():
-        values = g.op(
-            "Cast", values, to_i=_type_utils.JitScalarType.from_name(dtype).onnx_type()
+    try:
+        self_scalar_type = _type_utils.JitScalarType.from_value(self)
+        values_scalar_type = _type_utils.JitScalarType.from_value(
+            values, _type_utils.JitScalarType.UNDEFINED
         )
-    scalar_type = _type_utils.JitScalarType.from_name(dtype)
+        if self_scalar_type != values_scalar_type:
+            values = g.op("Cast", values, to_i=self_scalar_type.onnx_type())
+    except errors.OnnxExporterError:
```

Comment:
Possible to do away this try catch block as well, with

```python
if self_scalar_type == _type_utils.JitScalarType.UNDEFINED:
    ...
```

?

Reply:
done